src/bin/cli.rs

```diff
@@ -36,6 +35,25 @@ enum State {
     Decrypted(StateDecrypted),
 }
 
+impl Display for State {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let instruction = match self {
```

Comment:
I would suggest splitting these strings into two:
1. Is the output from `Display::fmt` and just gives a text representation of the state like `"Hi {name}, we just connected to server {url}.`
2. The possible action, can be implemented with another method like `State::fmt_action` and would be like `Please enter `next` to continue`

Reply:
Addressed in [05b80b8](https://github.com/ChihChengLiang/karma_calculator/pull/20/commits/05b80b86f60e7ee59a1561568892a6371a8facbc). Now display, update, and instructions are separated.